### DISCUSSION TITLE
local file system

### DISCUSSION THREAD

**Discussion by wipfli** [protomaps/PMTiles#137] [General]
Firefox seems to have some new file system api. Does this change anything how pmtiles can be used locally?

https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API

**Comment by bdon** [comment]
It's interesting, but what enhancements does it make available over https://developer.mozilla.org/en-US/docs/Web/API/File ?

Write support is interesting if we wanted to write a PMTiles locally, like extracting a larger archive to a smaller one.

**Comment by sachaw** [comment]
The File System Access API just adds utilities for us to manage traditional File objects in situ without having to have them loaded into your browser, this allows reading, writing, moving, renaming, folder enumeration etc.

Now a question, is it currently possible to pass a PMTile file (as a `File` object) and use it in MapLibre for example? All of the examples I have seen are using url's thanks.

**Reply by bdon** [reply]
yes, that's how the viewer works. see here:

https://github.com/protomaps/PMTiles/blob/main/app/src/Start.tsx#L111
https://github.com/protomaps/PMTiles/blob/main/app/src/MaplibreMap.tsx#L349

It's a little awkward because you need to add it via JavaScript first before it can be referenced in a style object.

**Reply by sachaw** [reply]
Many thanks!
